discovery_engine_core/core/src/engine.rs

```diff
@@ -48,15 +48,21 @@ pub enum Error {
     /// Invalid stack id: {0}.
     InvalidStackId(StackId),
 
-    /// An operation on a stack failed: {0}.
+    /// An operation on a stack failed with a stack error: {0}.
     StackOpFailed(#[source] stack::Error),
 
+    /// An operation on a stack failed with a generic error: {0}.
+    StackOpGen(#[source] GenericError),
+
     /// Error while selecting the documents to return: {0}.
     Selection(#[from] mab::Error),
 
     /// Error while using the ranker.
     Ranker(#[from] GenericError),
```

Comment:
We cannot use `from` here now that we have another error using that.

Reply:
I removed `StackOpGen` so the error variants here are back to the way they were